docsource/ops.rst

```diff
@@ -1,6 +1,228 @@
+
+In order to make it easy for the client to search across all indices of a certain type, aliases are applied with the following naming convention.
+
+.. code-block:: text
+
+ alias -> [indices]
+ genome -> [genome_1, genome_2 ... genome_n]
+
+When reindexing is necessary, the index is reindexed to a new index with a new search type version. The first reindexing operation performed on genome_1 will result in a new index genome_1_r1 that replaces genome_1 in the genome alias. n reindexing operations performed on genome_1 will result in a new index genome_1_rn that replaces index genome_1_rn-1 in the genome alias.
+
+Reindexing Process
+-------------------
+For the sake of simplicity and for the reasons described below. a single process has been defined for all reindexing cases -
+
+a) change field value
+b) add field type
+c) change field type
+d) remove field type
+
+The process involves reindexing an existing index into a new index for all of these cases. i.e. in-place reindexing is discouraged because the system may not have a recent snapshot/backup or any snapshot for recovery purposes should anything go wrong with the reindexing process.
+
+In addition, given that there can be as many as a thousand indices for KBase data, maintenance can become a challenge if the process is not simple. Some level of simplicity has been achieved here by defiining a single process that covers all the reindexing cases. If necessary, the process may be further simplified through some level of automation as it matures over time.
+
+.. note::
+
+    The commands below can be copy-pasted into Kibana and executed against the index. The corresponding curl commands can be obtained from Kibana by clicking on the little wrench icon that appears next to the pasted command.
+
+.. note::
+
+    If any of the steps below fail, don't proceed until the issue is resolved by referring to the ElasticSearch documentation.
+
+1. Note current time
+
+2. Stop any workers that are performing trickle updates on the index that needs to be reindexed.
+
+3. Refresh the index that needs reindexing to make sure it has been brought to a consistent state.
+
+.. code-block:: bash
+
+    POST /kbase.genome_1/_refresh
+
+4. Get a checksum for the index and record it in a separate file for later verification.
+
+.. code-block:: bash
+
+    GET /kbase.genome_1/_stats/docs,store
+
+5a. If mapping needs to be changed (for cases b, c, d above), get current index mapping.
+
+.. code-block:: bash
+
+    GET kbase.genome_1/_mapping
+
+5b. Copy-paste the mapping from the current index into the body section of the PUT command below and make the necessary field change (preferably one change per complete reindexing operation).
```

Comment:
Seems like a command for creating an index based on  a search type would reduce potential errors a lot.

Reply:
I agree. I have made a note of this for a future PR.